packages/curve-ui-kit/src/shared/ui/Metric.tsx

```diff
@@ -0,0 +1,193 @@
+
+import { SizesAndSpaces } from 'curve-ui-kit/src/themes/design/1_sizes_spaces'
+import { TypographyVariantKey, TYPOGRAPHY_VARIANTS } from 'curve-ui-kit/src/themes/typography'
+import { abbr, suffix } from 'curve-ui-kit/src/utils'
+
+const { Spacing } = SizesAndSpaces
+
+// Correspond to flexbox align items values.
+export const ALIGNMENTS = ['start', 'center', 'end'] as const
+type Alignment = (typeof ALIGNMENTS)[number]
+
+const MetricSize = {
+  small: 'highlightM',
+  medium: 'highlightL',
+  large: 'highlightXl',
+  extraLarge: 'highlightXxl',
+} as const satisfies Record<string, TypographyVariantKey>
+
+export const SIZES = Object.keys(MetricSize) as (keyof typeof MetricSize)[]
+
+type UnitOptions = {
+  symbol: string
+  position: 'prefix' | 'suffix'
+  abbreviate: boolean
+}
+
+const dollar: UnitOptions = {
+  symbol: '$',
+  position: 'prefix',
+  abbreviate: true,
+}
+
+const percentage: UnitOptions = {
+  symbol: '%',
+  position: 'suffix',
+  abbreviate: false,
+}
+
+export const UNITS = ['dollar', 'percentage'] as const
+type Unit = (typeof UNITS)[number] | UnitOptions
+
+const UNIT_MAP: Record<(typeof UNITS)[number], UnitOptions> = {
+  dollar,
+  percentage,
+} as const
+
+// Default value formatter.
+const formatValue = (value: number, decimals?: number): string => {
+  return value.toLocaleString(undefined, {
+    minimumFractionDigits: decimals,
+    maximumFractionDigits: decimals,
+  })
+}
+
+// Default notional value formatter.
+const formatNotionalValue = (value: number, decimals?: number): string => {
+  return value.toLocaleString(undefined, {
+    minimumFractionDigits: decimals,
+    maximumFractionDigits: decimals,
+  })
+}
+
+const formatChange = (value: number) => {
+  // Looks aesthetically more pleasing without decimals.
+  if (value === 0) return '0'
+
+  return value.toLocaleString(undefined, {
+    minimumFractionDigits: 2,
+    maximumFractionDigits: 2,
+  })
+}
+
+type Props = {
+  /** Label that goes above the value */
+  label: string
+
+  /** The actual metric value to display */
+  value: number
+  /** Optional formatter for metric value */
+  formatter?: (value: number) => string
+  /** If the value should be abbreviated to 1.23k or 3.45m */
+  abbreviate?: boolean
+  /** The number of decimals the value should contain */
+  decimals?: number
+  /** A unit can be a currency symbol or percentage, prefix or suffix */
+  unit?: Unit
+
+  /** Optional value that denotes a change in metric value since 'last' time */
+  change?: number
+
+  /** Notional values give extra context to the metric, like underlying value */
+  notional?: number
+  notionalFormatter?: (value: number) => string
+  notionalAbbreviate?: boolean
+  notionalDecimals?: number
+  notionalUnit?: Unit
+
+  size?: keyof typeof MetricSize
+  alignment?: Alignment
+  loading?: boolean
+}
+
+export const Metric = ({
+  label,
+
+  value,
+  formatter = (value: number) => formatValue(value, decimals),
+  abbreviate,
```

Comment:
maybe this should just be a formatter?

Reply:
You mean directly the contents of `formatValue` in this case? When I made this it wasn't a one-liner and didn't want to put a whole function definition as a default. Figured this would be a bit more clear? Not sure if you still want me to inline the function here.

Reply:
I think this comment is related to what you mentioned regarding the prop flattening.
Basically `formatter`, `abbreviate`, `unit` and `decimals` are 4 variables defining how to convert a number to string. Same for the notional value.
While flexibility is nice, it creates many different ways to use this component which can lead to inconsistency.
IMO we should have only one way to display dollars, amounts, percentages, etc
This is not a blocker to merge, though! We can always improve/change later.